.github/workflows/dependency_modification_check.yml

```
@@ -0,0 +1,18 @@
+name: Check no dependencies were modified
+
+on:
+  pull_request
```

Nit: The default trigger is `push` and I think this produces near identical behavior.
```suggestion
on:
  push
```
this will also run the action in the case we need to force-push to any branch like when rebasing a feature branch.

---

Seems like `push` doesnt trigger on pull_request. I've set it trigger on both now